sdk/src/main/scala/io/horizen/account/state/ProxyMsgProcessor.scala

```diff
@@ -0,0 +1,122 @@
+    ContractInteroperabilityFork.get(consensusEpochNumber).active
+  }
+
+  def doInvokeSmartContractStaticCallCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+    doInvokeSmartContractCmd(invocation, view, context, readOnly = true)
+  }
+
+  def doInvokeSmartContractCallCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+    doInvokeSmartContractCmd(invocation, view, context, readOnly = false)
+  }
+
+  private def doInvokeSmartContractCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext, readOnly : Boolean): Array[Byte] = {
+    log.debug(s"Entering with invocation: $invocation")
+
+    val value = invocation.value
+
+    // check that invocation.value is greater or equal than zero
+    if (value.signum() < 0) {
+      throw new ExecutionRevertedException("Value must not be zero")
+    }
+
+    // check that sender account exists
+    if (!view.accountExists(invocation.caller)) {
+      throw new ExecutionRevertedException(s"Sender account does not exist: ${invocation.caller}")
+    }
+
+    val inputParams = getArgumentsFromData(invocation.input)
+
+    val cmdInput = InvokeSmartContractCmdInputDecoder.decode(inputParams)
+    val contractAddress = cmdInput.contractAddress
+    val data = cmdInput.dataStr
+
+    if (view.isEoaAccount(contractAddress)) {
+      throw new ExecutionRevertedException(s"smart contract address is an EOA")
+    }
+
+    val dataBytes = Numeric.hexStringToByteArray(data)
+    val res = context.execute(
+      if (readOnly) {
+        log.debug(s"static call to smart contract, address=$contractAddress, data=$data")
+        invocation.staticCall(
+          contractAddress,
+          dataBytes,
+          invocation.gasPool.getGas // we use all the amount we currently have
+        )
+      } else {
+        log.debug(s"call to smart contract, address=$contractAddress, data=$data")
+        invocation.call(
+          contractAddress,
+          value,
+          dataBytes,
+          invocation.gasPool.getGas // we use all the amount we currently have
+        )
+      }
+    )
+
+    val proxyInvocationEvent = ProxyInvocation(invocation.caller, contractAddress, dataBytes)
+    val evmLog = getEthereumConsensusDataLog(proxyInvocationEvent)
+    view.addLog(evmLog)
+
+    // result in case of success execution might be useful for RPC commands
+    log.debug(s"Exiting with res: ${BytesUtils.toHexString(res)}")
+
+    res
+  }
+
+
+  @throws(classOf[ExecutionFailedException])
+  override def process(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+    log.debug(s"processing invocation: $invocation")
+    if (!isForkActive(context.blockContext.consensusEpochNumber)) {
+      throw new ExecutionRevertedException(s"fork not active")
+    }
+
+    if (params.isInstanceOf[MainNetParams]) {
```

Comment:
I have removed the checks on the chain network and on the fork point from `process()` because they were redundant.
I have added the check on the network in `canProcess()` method instead of `isForkActive()` because it is more correct there.